packages/workspace/src/browser/workspace-service.ts

```diff
@@ -68,9 +68,21 @@ export class WorkspaceService implements FrontendApplicationContribution {
 
     @postConstruct()
     protected async init(): Promise<void> {
-        const workspaceUri = await this.server.getMostRecentlyUsedWorkspace();
-        const workspaceFileStat = await this.toFileStat(workspaceUri);
-        await this.setWorkspace(workspaceFileStat);
+        let wpUriString: string | undefined = undefined;
+
+        // Prefer the workspace path specified as the URL fragment, if present.
+        if (window.location.hash.length > 1) {
+            // Remove the leading #.
+            const wpPath = window.location.hash.substring(1);
+            wpUriString = new URI().withPath(wpPath).withScheme('file').toString();
```

Comment:
Could we wrap line 74 - 77 into a separate helper function that has a descriptive name? It would make the code easier to unit test (`window` object is unavailable in unit tests).

Same comment applies to line 123-127.